tests/test_units/test_rms_context.py

```diff
@@ -155,34 +155,31 @@ def test_regsurf_metadata_with_timedata_legacy(rmssetup, rmsglobalconfig, regsur
         config=rmsglobalconfig,
         content="depth",
     )  # read from global config
-    meta1 = edata.generate_metadata(
-        regsurf,
-        name="TopVolantis",
-        timedata=[[20300101, "moni"], [20100203, "base"]],
-    )
-    logger.debug(prettyprint_dict(meta1))
+
+    # should raise userwarning
+    with pytest.warns(UserWarning):
+        meta1 = edata.generate_metadata(
+            regsurf,
+            name="TopVolantis",
+            timedata=[[20300101, "moni"], [20100203, "base"]],
+        )
+
     assert "topvolantis--20300101_20100203" in meta1["file"]["relative_path"]
 
-    assert meta1["data"]["time"][0]["value"] == "2010-02-03T00:00:00"
-    assert meta1["data"]["time"][0]["label"] == "base"
-    assert meta1["data"]["time"][1]["value"] == "2030-01-01T00:00:00"
-    assert meta1["data"]["time"][1]["label"] == "moni"
+    # new format should be present in the metadata files
+    assert meta1["data"]["time"]["t0"]["value"] == "2010-02-03T00:00:00"
+    assert meta1["data"]["time"]["t0"]["label"] == "base"
+    assert meta1["data"]["time"]["t1"]["value"] == "2030-01-01T00:00:00"
+    assert meta1["data"]["time"]["t1"]["label"] == "moni"
 
-    meta1 = edata.generate_metadata(
+    # check that metadata are equal independent of legacy_time_format
+    dataio.ExportData.legacy_time_format = False
```

Comment:
Should this be True, not False? Also, should call to generate_metadata below be wrapped in a check for the produced warning?

Reply:
No it is meant as is 🙂 The first call to create metadata is with the option `legacy_time_format=True`, and that one is wrapped in the `pytest.warn`. This second call is with the option `legacy_time_format=False` and then it is checked that the metadata produced for `data.time` are equal to the first run. I was actually going to check the `data` block all together, but the test failed leading me to find a bug #558 👍